src/eventbridge/eventpattern/index.ts

```diff
@@ -0,0 +1,798 @@
+import { FunctionDecl } from "../../declaration";
+import {
+  BinaryExpr,
+  CallExpr,
+  ElementAccessExpr,
+  Expr,
+  PropAccessExpr,
+  UnaryExpr,
+} from "../../expression";
+import {
+  assertDefined,
+  assertNever,
+  assertNumber,
+  assertString,
+} from "../../assert";
+import { isReturn, Stmt } from "../../statement";
+import {
+  PatternDocument,
+  Pattern,
+  isPatternDocument,
+  isNumericAggregationPattern,
+  isNumericRangePattern,
+  isPresentPattern,
+  isAggregatePattern,
+  isAnythingButPattern,
+  isExactMatchPattern,
+  isPrefixMatchPattern,
+  isEmptyPattern,
+  patternDocumentToEventPattern,
+} from "./pattern";
+import * as fnls_event_bridge from "./types";
+import {
+  joinNumericRange,
+  reduceNumericAggregate,
+  mergeNumericOr,
+  negateNumericRange,
+  andNumericAggregation,
+  andNumericAggregationWithRange,
+  createSingleNumericRange,
+} from "./numeric";
+import {
+  BinaryOp,
+  isBinaryExpr,
+  isBooleanLiteral,
+  isCallExpr,
+  isElementAccessExpr,
+  isIdentifier,
+  isNullLiteralExpr,
+  isNumberLiteralExpr,
+  isPropAccessExpr,
+  isStringLiteralExpr,
+  isUnaryExpr,
+} from "../..";
+import {
+  INCLUDES_SEARCH_ELEMENT,
+  OPERATIONS,
+  STARTS_WITH_SEARCH_STRING,
+} from "./constants";
+import { invertBinaryOperator } from "./utils";
+
+/**
+ * https://github.com/sam-goodwin/functionless/issues/37#issuecomment-1066313146
+ */
+export const synthesizeEventPattern = (
+  predicate: FunctionDecl
+): fnls_event_bridge.FnLsEventPattern => {
+  const eventDecl = predicate.parameters[0];
+
+  const handleExpr = (expr: Expr): PatternDocument => {
```

Comment:
is `eval` a better name than `handle`? We should try and align the terminology in each of the interpreters.

Reply:
I can switch to eval.